script/install.sh

```diff
@@ -591,15 +586,23 @@ install_compiler() {
         if [[ $ARCH != x86* ]]; then
             fatal "no DMD binaries available for $ARCH"
         fi
-        local url="http://downloads.dlang.org/nightlies/$1/$basename.tar.xz"
+        local ext
+        test $OS = windows && ext=.7z || ext=.tar.xz
```

Comment:
Why can't we use the existing `ext`?

Reply:
You mean declare `ext` once at the top of the function instead of per `if` branch?